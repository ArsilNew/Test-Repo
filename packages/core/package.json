{
  "name": "@kenshooui/react-multi-select",
  "version": "2.0.0",
  "description": "React Multi Select",
  "main": "dist/react-multi-select.cjs.js",
  "module": "dist/react-multi-select.esm.js",
  "repository": "https://github.com/kenshoo/react-multi-select.git",
  "author": "Kenshoo",
  "license": "MIT",
  "private": false,
  "peerDependencies": {
    "react": "^16.13.1"
  },
  "devDependencies": {
    "react": "^16.13.1"
  }
}
