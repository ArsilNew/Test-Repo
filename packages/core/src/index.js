import React from "react";

const ReactMultiSelect = () => <div>Hello World 2</div>;

    export default ReactMultiSelect;