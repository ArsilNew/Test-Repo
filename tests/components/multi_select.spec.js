import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { mount } from "enzyme";

import MultiSelectWithState, {
  MultiSelect
} from "../../src/components/multi_select";
import { SearchWithValue } from "../../src/components/search/search";
import { createGenerateClassName } from "@material-ui/core/styles";

const generateClassName = createGenerateClassName({
  dangerouslyUseGlobalCSS: true,
  productionPrefix: "c"
});

const CustomComponent = jest
  .fn(() => <div>Custom Component</div>)
  .mockName("mockedComponent");

const custom_messages = {
  searchPlaceholder: "Find...",
  noItemsMessage: "No entries available...",
  noneSelectedMessage: "Nothing",
  selectedMessage: "Checked",
  selectAllMessage: "Check all",
  clearAllMessage: "Uncheck all"
};

const selectAllItems = jest.fn().mockName("selectAllItems");
const filterItems = jest.fn().mockName("filterItems");
const selectItem = jest.fn().mockName("selectItem");
const unselectItems = jest.fn().mockName("unselectItems");
const clearAll = jest.fn().mockName("clearAll");

describe("MultiSelect", () => {
  test("default snapshot", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect />);
    expect(tree).toMatchSnapshot();
  });

  test("custom selectAllRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect selectAllRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom searchRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect searchRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom noItemsRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect noItemsRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom itemRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect itemRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom listRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect listRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom messages", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect messages={custom_messages} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedIds", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect selectedIds={[1, 2]} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect selectedItems={[1, 2]} />);
    expect(tree).toMatchSnapshot();
  });

  test("passes disabled if maxSelectedItem has passed", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect selectedItems={[1, 2]} maxSelectedItems={2} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("does not pass disabled if maxSelectedItem has passed", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect selectedItems={[1, 2]} maxSelectedItems={4} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("passed selectAllItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect selectAllItems={selectAllItems} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("passed filterItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect selectAllItems={filterItems} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectItem", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect selectAllItems={selectItem} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed unselectItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect unselectItems={unselectItems} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed clearAll", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect clearAll={clearAll} />);
    expect(tree).toMatchSnapshot();
  });

  test("can remove select all", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect showSelectAll={false} />);
    expect(tree).toMatchSnapshot();
  });

  test("can remove search", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect showSearch={false} />);
    expect(tree).toMatchSnapshot();
  });

  test("displays Loader when loading", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect loading={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("displays custom Loader when loading", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect loading={true} loaderRenderer={CustomComponent} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("can remove selected items", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect showSelectedItems={false} />);
    expect(tree).toMatchSnapshot();
  });

  test("will pass itemHeight", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect itemHeight={60} />);
    expect(tree).toMatchSnapshot();
  });

  test("will pass selectedItemHeight", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect selectedItemHeight={60} />);
    expect(tree).toMatchSnapshot();
  });

  test("will pass selectedItemHeight", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect itemHeight={40} selectedItemHeight={60} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("will pass selectAllHeight", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect itemHeight={60} selectAllHeight={40} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("will pass selectAllHeight without itemHeight", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelect selectAllHeight={20} />);
    expect(tree).toMatchSnapshot();
  });

  test("searchValueChanged will be called on controlled search value", () => {
    const newValueString = "new value";
    const searchValueChanged = jest.fn();
    const tree = mount(
      <MultiSelectWithState
        searchRenderer={SearchWithValue}
        searchValue=""
        searchValueChanged={searchValueChanged}
        searchIcon="div"
      />
    );

    tree
      .find('input[type="text"]')
      .simulate("change", { target: { value: newValueString } });
    expect(searchValueChanged).toHaveBeenCalledWith(newValueString);
  });

  test("without responsiveHeight", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<MultiSelectWithState />);
    expect(tree).toMatchSnapshot();
  });
  test("with responsiveHeight", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelectWithState responsiveHeight={"70%"} />
    );
    expect(tree).toMatchSnapshot();
  });

  test("with generateClassName", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <MultiSelect generateClassName={generateClassName} />
    );
    expect(tree).toMatchSnapshot();
  });
});
