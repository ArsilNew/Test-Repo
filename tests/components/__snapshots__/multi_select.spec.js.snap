// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelect can remove search 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={360}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={false}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect can remove select all 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={355}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect can remove selected items 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
</div>
`;

exports[`MultiSelect custom itemRenderer 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[MockFunction mockedComponent]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect custom messages 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={
      Object {
        "clearAllMessage": "Uncheck all",
        "noItemsMessage": "No entries available...",
        "noneSelectedMessage": "Nothing",
        "searchPlaceholder": "Find...",
        "selectAllMessage": "Check all",
        "selectedMessage": "Checked",
      }
    }
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={
      Object {
        "clearAllMessage": "Uncheck all",
        "noItemsMessage": "No entries available...",
        "noneSelectedMessage": "Nothing",
        "searchPlaceholder": "Find...",
        "selectAllMessage": "Check all",
        "selectedMessage": "Checked",
      }
    }
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect custom noItemsRenderer 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[MockFunction mockedComponent]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[MockFunction mockedComponent]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect custom searchRenderer 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[MockFunction mockedComponent]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect custom selectAllRenderer 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[MockFunction mockedComponent]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect default snapshot 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect displays Loader when loading 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <Loader />
</div>
`;

exports[`MultiSelect displays custom Loader when loading 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <mockConstructor />
</div>
`;

exports[`MultiSelect does not pass disabled if maxSelectedItem has passed 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={360}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    showSearch={true}
    showSelectAll={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    selectedItemRenderer={[Function]}
    selectedItems={
      Array [
        1,
        2,
      ]
    }
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect passed clearAll 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={[MockFunction clearAll]}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect passed filterItems 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={[MockFunction filterItems]}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect passed selectAllItems 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={[MockFunction selectAllItems]}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect passed selectItem 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={[MockFunction selectItem]}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect passed selectedIds 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect passed selectedItems 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    selectedItemRenderer={[Function]}
    selectedItems={
      Array [
        1,
        2,
      ]
    }
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;

exports[`MultiSelect passed unselectItems 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={315}
    disabled={false}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={Array []}
    showSearch={true}
    showSelectAll={true}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={Array []}
    selectedItemRenderer={[Function]}
    selectedItems={Array []}
    selectionStatusRenderer={[Function]}
    unselectItems={[MockFunction unselectItems]}
  />
</div>
`;

exports[`MultiSelect passes disabled if maxSelectedItem has passed 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "height": 400,
    }
  }
>
  <SourceList
    calculatedHeight={360}
    disabled={true}
    filterItems={undefined}
    filteredItems={Array []}
    getList={undefined}
    isAllSelected={false}
    itemHeight={40}
    itemRenderer={[Function]}
    messages={Object {}}
    noItemsRenderer={[Function]}
    onSearchChange={undefined}
    searchIcon={undefined}
    searchRenderer={[Function]}
    selectAllItems={undefined}
    selectAllRenderer={[Function]}
    selectItem={undefined}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    showSearch={true}
    showSelectAll={false}
  />
  <DestinationList
    clearAll={undefined}
    height={400}
    itemHeight={40}
    messages={Object {}}
    noItemsRenderer={[Function]}
    selectedIds={
      Array [
        undefined,
        undefined,
      ]
    }
    selectedItemRenderer={[Function]}
    selectedItems={
      Array [
        1,
        2,
      ]
    }
    selectionStatusRenderer={[Function]}
    unselectItems={undefined}
  />
</div>
`;
